plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -630,10 +639,10 @@ private static boolean isManagedTable(Table table)
         return table.getTableType().equals(MANAGED_TABLE.name());
     }
 
-    private static void deleteDir(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path, boolean recursive)
+    private void deleteDir(HdfsContext context, Location path)
```

Comment:
This causes minor behavior change. For instance, we can't delete the directory having `#` anymore. I suppose this is acceptable and we should deny creating such a database or table.

Reply:
I assume this has no real impact since for example if partition columns include `#` (in their name or value) the directory name contains it in URL encoded form and not raw `#`.